packages/worker/src/worker-options.ts

```diff
@@ -75,9 +75,9 @@ export interface WorkerOptions {
   shutdownSignals?: NodeJS.Signals[];
 
   /**
-   * TODO: document, figure out how to propagate this to the workflow isolate
+   * Path of a module with a `dataConverter` named export. `dataConverter` should be an instance of a class that implements {@link DataConverter}.
    */
-  dataConverter?: DataConverter;
+  dataConverterPath?: string;
```

Comment:
Looks like a breaking change here, but if needed I understand.

Do we _have_ to use the data converter _inside_ the workflow VM isolate? One wonders if a data converter should have a proxy to the main node process and use the data converter there. And if we don't have to use it inside the isolate, can we leave the interface as it was?

Reply:
It's breaking a feature that's not fully implemented yet (currently `defaultDataConverter` is used for the Workflow, disregarding the `dataConverter` option).

We can't send functions, including those in `ProtoActivityInput` or `ProtoMessage` instances, from the main thread to the worker thread—just [basic builtin JS types](https://nodejs.org/api/worker_threads.html#portpostmessagevalue-transferlist). If we used `ProtoMessage`, we could deserialize in main thread to an intermediate format, and then convert to `ProtoMessage` inside the worker thread. But our intermediate format wouldn't support an arbitrary custom data converter. So in order to allow custom data converters, the converting code needs to run in the worker thread.

Reply:
That makes sense. It's unfortunate that due to how protobufjs works, it creates classes with methods instead of simple JSON representations. If we had the requirement of "all items in and out of data converters must be trivially JSON serializable, all functions and prototype pieces do not cross the boundary" then we could just serialize on one side. But then you couldn't know whether it was originally a proto message.

I guess technically the encoding could be a key on the object (e.g. `T & { _encoding: string }`) but that has many issues as well. I already pitied TS devs who were forced to bundle workflows separately (can't have code side-by-side like our other SDKs), and now this has leaked into data converters too. But the limitations definitely make sense :+1: